src/components/engagement/engagement.repository.ts

```diff
@@ -28,4 +29,30 @@ export class EngagementRepository {
       .run();
     return rows.map((r) => r.id);
   }
+
+  async rolesInScope(engagementId: string, session: Session) {
+    const query = this.db
+      .query()
+      .match([
+        node('eng', 'Engagement', { id: engagementId }),
+        relation('in', 'engagement', { active: true }),
+        node('node', 'Project'),
+        relation('out', '', 'member', { active: true }),
+        node('projectMember', 'ProjectMember'),
+        relation('out', '', 'user', { active: true }),
+        node('user', 'User', { id: session.userId }),
+      ])
+      .match([
+        node('projectMember'),
+        relation('out', 'r', 'roles', { active: true }),
+        node('roles', 'Property'),
+      ])
+      .return('apoc.coll.flatten(collect(roles.value)) as memberRoles')
```

Comment:
What if a project member has multiple roles? Would `collect(roles.value)` yield an array of arrays? I don't remember if project members have a list for the value of roles if they have more than one role.

Reply:
Hence the `apoc.coll.flatten`call

Reply:
Does it need to be recursive if `collect(roles.value) = [[],[],[]]`?

https://neo4j.com/labs/apoc/4.0/overview/apoc.coll/apoc.coll.flatten/

Reply:
No because we only need to flatten one level.